flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -206,7 +207,10 @@ public RestClusterClient(
                 clusterId,
                 new ExponentialWaitStrategy(10L, 2000L),
                 factory,
-                Files.createTempDirectory("flink-rest-client-jobgraphs"));
+                Files.createDirectories(
+                        Paths.get(
+                                ConfigurationUtils.parseTempDirectories(config)[0],
+                                "flink-rest-client-jobgraphs")));
```

Comment:
Looks to me there's no need to create an additional folder here. 
If the submission succeeds, job graph files will be cleaned up, while leaving this folder remained in the tmp dir.